### PR TITLE
[Communication] - SDK - Synced SMS and Phone Numbers shared folders

### PR DESCRIPTION
**Pull request by jbeauregardb**
SMS and Phone Numbers packages had an outdated version of their shared folders. Identity and chat were updated so I replicated the changes on these two other SDKs.

### CODE REVIEW

**Review by DominikMe — DISMISSED** [review]
Thanks Jorge!

We also need to update setup.py to add typing-extensions as a dependency. See @annatisch 's PR as an example: https://github.com/Azure/azure-sdk-for-python/pull/17912

**Comment by annatisch** [comment]
@jbeauregardb - there are other files in `_shared` that have changed since these were last synced. I think `chat` is the only SDK with the latest code. Have you tried copying over the entire directory?

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_shared/models.py [hunk 1]
```diff
@@ -6,210 +6,151 @@
...
 from enum import Enum, EnumMeta
 from six import with_metaclass
+from typing import Mapping, Optional, Union, Any
+try:
+    from typing import Protocol, TypedDict
+except ImportError:
+    from typing_extensions import Protocol, TypedDict
 
-import msrest
+from azure.core import CaseInsensitiveEnumMeta
```
Comment (annatisch) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_shared/models.py:15]: You will need to also bump the `azure-core` dependency to ensure `CaseInsensitiveEnumMeta` is present.

**Review by DominikMe — COMMENTED** [review]
Unblocking my review to just Comment. Anna has this and you don't need my approval :-)

**Review by DominikMe — COMMENTED** [review]
.

**Review by annatisch — APPROVED** [review]

**Pull request merged**